### PR TITLE
Azure Functions isolated process

### PR DESCRIPTION
**Pull request by cajuncoding**
Summary of the changes (Less than 80 chars)

- Implemented support for Azure Functions Isolated process model
- Expanded and added Unit Test coverage for both In-process & Isolated process models

### CODE REVIEW

**Comment by michaelstaib** [comment]
I moved it to the main branch ... we cherry-pick the change to version 12 once we have merged it.

File: src/HotChocolate/AzureFunctions/src/HotChocolate.AzureFunctions.IsolatedProcess/HotChocolate.AzureFunctions.IsolatedProcess.csproj [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <PackageId>HotChocolate.AzureFunctions.IsolatedProcess</PackageId>
+    <AssemblyName>HotChocolate.AzureFunctions.IsolatedProcess</AssemblyName>
+    <RootNamespace>HotChocolate.AzureFunctions.IsolatedProcess</RootNamespace>
+    <Description>This package contains the GraphQL AzureFunctions Isolated Process integration for Hot Chocolate.</Description>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Remove="Extensions\HotChocolateAzFuncIsolatedProcessServiceCollectionExtensions.cs" />
```
Comment (michaelstaib) [inline src/HotChocolate/AzureFunctions/src/HotChocolate.AzureFunctions.IsolatedProcess/HotChocolate.AzureFunctions.IsolatedProcess.csproj:13]: Is this an accident?

Comment (cajuncoding) [reply src/HotChocolate/AzureFunctions/src/HotChocolate.AzureFunctions.IsolatedProcess/HotChocolate.AzureFunctions.IsolatedProcess.csproj:13]: I never got a notification for this comment, just now seeing it :-/ 

Working on getting my build working and I'll validate, but def. seems like an accident...

Comment (cajuncoding) [reply src/HotChocolate/AzureFunctions/src/HotChocolate.AzureFunctions.IsolatedProcess/HotChocolate.AzureFunctions.IsolatedProcess.csproj]: Ok this is resolved, it was def. an accident somehow. ✅

**Comment by cajuncoding** [comment]
@michaelstaib someone on slack asked a good question this evening about HttpContext support via IHttpContextAccessor... I used one of my local demo projects to quickly test that it works as expected. 

But I didn't add support for that in the new Isolated process model since we control the `HttpContext` creation, so I'm adding that and will push an update soon!  It should work just like the existing Documentation shows 👍.

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**